crates/tauri-plugin-holochain-service/android/src/main/java/com/plugin/holochain_service/HolochainService.kt

```diff
@@ -131,6 +150,81 @@ class HolochainService : Service() {
             installedAppId: String
         ) {
             Log.d(TAG, "ensureAppWebsocket")
+            this.expectAuthorized()
+
+            serviceScope.launch(Dispatchers.IO) {
+                callback.ensureAppWebsocket(AppAuthFfiParcel(runtime?.ensureAppWebsocket(installedAppId)!!))
+            }
+        }
+
+        /// Sign a zome call
+        override fun signZomeCall(
+            callback: IHolochainServiceCallback,
+            req: ZomeCallUnsignedFfiParcel
+        ) {
+            Log.d(TAG, "signZomeCall")
+            this.expectAuthorized()
+
+            serviceScope.launch(Dispatchers.IO) {
+                callback.signZomeCall(ZomeCallFfiParcel(runtime!!.signZomeCall(req.inner)))
+            }
+        }
+
+        // We cannot call Binder.getCallingUid() within the onBind callback,
+        // so instead we check the authorization within each IPC call
```

Comment:
Originally I tried to implement the authorization gate in the `onBind` callback -- i.e. only authorized packages could bind to the service.

That didn't work because `onBind` is actually called by the android system, so there was no way of finding out which package had requested the binding.

Instead we always allow the binding, but then have the authorization gate within each individual binding function.